### ISSUE TITLE
Mask painting -- memory save, updating, writing

### ISSUE THREAD

**Issue by thjarrett** [idia-astro/iDaVIE#106]
Painting masks can be a time-consuming experience; as such, it is important that we make it easy and safe for user to not lose the information they input (painting a mask).   So I would request that the newly painted mask be saved into memory (regardless of whether it is formally saved or not).   Give the user the ability to *clear* the new masking, but otherwise retain it even when leaving the painting mode.  The user should then have the ability to SAVE the mask (write to new mask) outside of paint mode, as an option (say in the quick menu).  

Related:  if a mask has been modified (see above), then when the user leaves iDAVIE -- either to quit or to load a different cube/or/mask,  then a window should pop up, asking the user if they want to SAVE the new mask.   That way, work is not lost (because of the user may have forgotten to save the mask). 

So to summarize:
1.  save the mask in memory automatically, even if you dont' write the new mask and leave paint mode
2.  allow the user to write the new mask, outside of paint mode
3.  remind the user that they have modified the mask, to either save it or jettison, when they are either leaving idavie or loading a new cube mask (or a new cube that is not relevant to the mask in memory).
